src/website/webview/views.py

```diff
@@ -518,103 +518,89 @@ def get_queryset(self) -> Any:
-            update_suggestion(request, filter=filter_for_suggestions)
+        # We want to provide graceful fallback for important workflows, when users have JavaScript disabled
+        js_enabled: bool = "no-js" not in request.POST
+        undo_status_change: bool = "undo-status-change" in request.POST
+        suggestion_id = request.POST.get("suggestion_id")
+        new_status = request.POST.get("new_status")
+        current_page = request.POST.get("page", "1")
+        suggestion = get_object_or_404(CVEDerivationClusterProposal, id=suggestion_id)
+        activity_log = SuggestionActivityLog(suggestion=suggestion).get_structured_log()
+        cached_suggestion = get_object_or_404(
+            CachedSuggestions, proposal_id=suggestion_id
         )
 
-        if new_status == "REJECTED":
+        # We only have to modify derivations when they are editable
+        if not (
+            self.status_filter == CVEDerivationClusterProposal.Status.REJECTED
+            or undo_status_change
+        ):
+            selected_derivations = [
+                str.split(",") for str in request.POST.getlist("derivation_ids")
+            ]
+            selected_derivations = set(map(int, chain(*selected_derivations)))
+            # We only allow for removal of derivations here, not for additions
+            derivation_ids_to_keep = set(
+                suggestion.derivations.filter(id__in=selected_derivations).values_list(
+                    "id", flat=True
+                )
+            )
+            suggestion.derivations.set(derivation_ids_to_keep)
+
+            # TODO: this is quite slow and bad.
+            # we are getting the JSON here and then sending it back.
+            # a more optimal way to do it is to perform the raw SQL query directly on pgsql
+            # something along the lines of:
+            # UPDATE SET payload = payload -# {an list of indices to remove contained in this list} WHERE proposal_id = proposal_id
+            # the problem is that computing the list of indices is pretty hard.
+            # this seems to encourage to move the payload format to an dict of derivation id → derivation contents
+            # this way, we already know which IDs to remove.
+            # this is left as future work.
+            new_packages = {
+                pname: v
+                for pname, v in cached_suggestion.payload["packages"].items()
+                if any(did in selected_derivations for did in v["derivation_ids"])
+            }
+            cached_suggestion.payload["packages"] = new_packages
+            cached_suggestion.save()
+
+        if new_status == "rejected":
             suggestion.status = CVEDerivationClusterProposal.Status.REJECTED
-        elif new_status == "ACCEPTED":
+        elif new_status == "accepted":
             suggestion.status = CVEDerivationClusterProposal.Status.ACCEPTED
+        elif new_status == "pending" and undo_status_change:
```

Comment:
This ensures, that there is only one way back to the pending state: By clicking the undo button. So this is really only there for double-checking anyway, as there is currently no button to achieve this.

Reply:
```suggestion
        # there's no UI for returning a suggestion back to pending state
        # but this is an additional safeguard to ensure it it won't happen
        #                                vvvvvvvvvvvvvvvvvv
        elif new_status == "pending" and undo_status_change:
```

Reply:
Please always answer why-questions in comments, otherwise that knowledge will be lost, and eternal cargo cult ensues (severely exerbated by not having tests)

Reply:
Okay, makes sense, I will try to do that :+1: